src/sprites/Elevator.lua

```diff
@@ -72,118 +66,225 @@ function Elevator:collisionResponse(_)
 end
 
---- Used specifically for when jumping while moving up with elevator.
-function Elevator:disableCollisionsForFrame()
-  self:setCollisionsEnabled(false)
+--
+--
+----- Private class methods
+--
 
-  self.isCollisionsDisabledForFrame = true
-end
+local function getActivationMovement(self, key)
+  if self.fields.orientation == ORIENTATION.Horizontal then
+    -- Horizontal orientation, return positive if Right, negative if Left
 
--- Private class methods
+    if key == KEYNAMES.Right then
+      return self.speed
+    elseif key == KEYNAMES.Left then
+      return -self.speed
+    end
+  else
+    -- Vertical orientation, return positive if Down, negative if Up
 
---- Get remaining movement based on direction and displacement
-local function getMovementRemaining(self, key)
-  -- Get inverted direction boolean
-  local isInverseDirection = isInverseDirection(key)
-
-  -- Remaining displacement - either towards displacementEnd or displacementStart if inverse direction
-  local displacementRemaining = isInverseDirection 
-    and self.displacement - self.displacementStart
-    or self.displacementEnd - self.displacement
-  
-  -- Calculate movement as scalar value, invert speed if inverse direction
+    if key == KEYNAMES.Down then
+      return self.speed
+    elseif key == KEYNAMES.Up then
+      return -self.speed
+    end
+  end
 
-  return math.min(displacementRemaining, self.speed) * (isInverseDirection and -1 or 1)
-end
+  -- If key is not a correct activation key, return 0
 
--- Public class Methods
+  return 0
+end
 
---- Sets movement to be executed in the next update() call using vector.
---- *param* key - the player input key direction (KEYNAMES)
---- *returns* whether an activation occurred based on key press.
-function Elevator:activate(key)
-  local movement = 0
-
-  if (key == KEYNAMES.Left or key == KEYNAMES.Right)
-  and self.fields.orientation == ORIENTATION.Horizontal then
-    -- Horizontal movement - get distance remaining
-    movement = getMovementRemaining(self, key)
-
-  elseif (key == KEYNAMES.Down or key == KEYNAMES.Up)
-  and self.fields.orientation == ORIENTATION.Vertical then
-    -- Vertical movement - get distance remaining
-    movement = getMovementRemaining(self, key)
+--- Get remaining movement based on direction and displacement
+local function getMovementRemaining(self, movement)
+  if movement < 0 then
+    return math.max(-self.displacement, movement)
+  elseif movement > 0 then
+    return math.min(self.displacementEnd - self.displacement, movement)
+  else
+    return 0
   end
-  
-  -- Set movement update scalar
-  self.movement = movement
+end
 
-  self.displacement += self.movement * _G.delta_time
+--- Checks collision for frame, also checking if child collides. Returns a partial movement for itself
+--- if elevator or child collides with another object.
+local function checkIfCollides(self, idealX, idealY, spriteToCheck)
+  local spriteToCheck = spriteToCheck or self
```

Comment:
could we use a different name for the arg and the local variable

Reply:
This was done to allow for recursion. Also it's common in lua to "or" an argument i.e. provide a default value.

Reply:
Added in the recursion :D